08_leukemia_cell_lines/workflow/workflow.sh

```diff
@@ -0,0 +1,370 @@
+#!/bin/bash
+##
+## Giulia Moro
+##
+## Fusion analysis for the leukemia dataset
+##
+
+## this script is to be run after the rock_roi_method workflow: https://github.com/imallona/rock_roi_method/tree/main
+# only on TSO data, since the fusions are expected to be there
+
+# STAR path 
+
+export PATH=/home/imallona/soft/star/STAR-2.7.10b/source:$PATH
+
+# for STARsolo
+
+COMBINED_FA_GENOME=~/mapping_leukemia_data/genome/combined.fa
+COMBINED_GTF_GENOME=~/mapping_leukemia_data/genome/combined.gtf
+WD=~/cell_lines_leukemia/rock_roi_repo_for_cell_lines/workflow_aln
+COMBINED_INDEXED_GENOME=~/mapping_leukemia/data/index
+STARSOLO_BAM=/home/gmoro/cell_lines_leukemia/rock_roi_repo_for_cell_lines/output/align_tso/leukemia_cell_lines/Aligned.sortedByCoord.out.bam
+
+# for bwa mem2
+
+TRANSCRIPTOME=gencode.v38.pc_transcripts.fa
+
+CUSTOM_FA=~/leukemia_bwamem2/genome/BCR_ABL.fa
+CUSTOM_GTF=~/leukemia_bwamem2/genome/BCR_ABL.gtf
+
+# for STAR fusion
+
+STAR_FUSION_GENOME=~/test_starfusion/genome/ctat_genome_lib_build_dir
+PATH_STAR_FUSION=~/STAR-Fusion/ctat-genome-lib-builder
+PATH_SIMG_FILE=~/star-fusion.v1.13.0.simg
+
+# threads
+
+NTHREADS=30
+
+# get unmapped reads and reads that mapped to BCR and to ABL and append their name to the readname
+
+## Step 1: get unmapped reads, reads in the BCR ABL slice, reads with no gx tag from .bam file and output in new .fastq file. Append the CB and UB tag to their read id. Deduplicate CB and UB.  
+
+cd $WD
+
+# getting reads in BCR and ABL region and adding 1M bp to each side
+
+bcr_start=23179704
+bcr_end=23318037
+abl_start=130713016
+abl_end=130887675
+
+echo "getting read names and deduplicating"
+
+# deduplication is done based on first sorting by UB (28), CB (27) and 1 (read id). If the combination of UB and CB is unique as well as the read name, then the read is appended. 
+# if first if loop check if read is already present in .fastq file or not, otherwise not added. 
+
+samtools view -H $STARSOLO_BAM -@ $NTHREADS > header.txt
+
+samtools view "$STARSOLO_BAM" "chr22:22179704-24318037" -@ "$NTHREADS" \
```

Comment:
These awk onelines are hard to parse, should I run them on the simulated/"known-truth" data? if so, how? I think you probably have another script to call this script on the simulated data and on "real" data, but haven't managed to find it. Or is it the Snakefile?